plugins/BEdita/Core/src/Filesystem/FilesystemAdapter.php

```diff
@@ -0,0 +1,107 @@
+<?php
+/**
+ * BEdita, API-first content management framework
+ * Copyright 2017 ChannelWeb Srl, Chialab Srl
+ *
+ * This file is part of BEdita: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * See LICENSE.LGPL or <http://gnu.org/licenses/lgpl-3.0.html> for more details.
+ */
+
+namespace BEdita\Core\Filesystem;
+
+use Cake\Core\InstanceConfigTrait;
+use League\Flysystem\AdapterInterface;
+
+/**
+ * Filesystem adapter.
+ *
+ * This abstract class is a wrapper around Flysystem adapters, to ensure they can be instantiated
```

Comment:
Yup, Flysystem is actually the name of the abstraction library we use under the hood. Can be misleading at times. 🙃